solr/core/src/java/org/apache/solr/security/PKIAuthenticationPlugin.java

```diff
@@ -377,8 +376,8 @@ PublicKey fetchPublicKeyFromRemote(String nodename) {
 
   @Override
   public void setup(Http2SolrClient client) {
-    final HttpListenerFactory.RequestResponseListener listener =
-        new HttpListenerFactory.RequestResponseListener() {
+    final var listener =
```

Comment:
this listener no per-request state so can simply add this listener to client level.